src/codecs/avif/decoder.rs

```diff
@@ -158,6 +158,11 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for AvifDecoder<R> {
             }
         }
 
+        // Convert Bgra to Rgba
+        for chunk in buf.chunks_exact_mut(4) {
+            chunk.swap(0, 2);
+        }
```

Comment:
I would like to see this handled in `DynamicImage` instead, if possible. The decision to remove `Bgra` would then be local to our buffer types while more complex third-parties can make use of the original byte order.

Reply:
I'd say that the correct thing would be to report `Bgra` as the `original_color_type()`, and then a forthcoming API addition could expose the ability for advanced users to decode images with the original color type instead of the more constrained `color_type()`. Keeping `ColorType::Bgra` means that _every_ user of ImageDecoder has to at least in theory handle BGRA images.

The library actually isn't even currently consistent on this area. The TGA decoder actually [already swaps BGR -> RGB](https://github.com/image-rs/image/blob/7d280f5745fcbd1c8af819b265dc81b63cacd6b4/src/codecs/tga/decoder.rs#L318).